packages/ui-kit/src/helpers/graphql/stories/DataPoolInput.stories.tsx

```diff
@@ -0,0 +1,26 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import type { DataPoolInput } from '../index'
+
+export const DataPoolInputComponent: React.FC<DataPoolInput> = () => null
+
+const meta = {
+  title: 'API/DataPoolInput',
+  tags: ['pattern'],
+  component: DataPoolInputComponent,
+  argTypes: {
+    id: {
+      control: 'text',
+      description: 'The id of the data pool to use for the metric. Required if `name` is not provided.'
+    },
+    name: {
+      control: 'text',
+      description: 'The name of the data pool to use for the metric. Required if `id` is not provided.'
+    }
+  }
```

Comment:
```suggestion
```
We should upload the descriptions from the type definitions.

Reply:
I think that is not possible since the graphql generator generates the comments just above the object not props for the `@oneOf` declarations. 
when you remove argsType from story you will be no able to see the desc. You can try it.

Reply:
Right. I think we should show `@oneOf` differently then. Currently, it looks like it's possible to pass more than one prop. Let's brainstorm it

Reply:
Yeah, I am curious if at least `type u = { x: string } | { y: string }` renders better than `type u = { x: string, y: never } | { x: never, y: string }`.

Otherwise, including a note that only one should be provided could be OK.

Other options?